src/openrct2-ui/windows/Scenery.cpp

```diff
@@ -684,6 +719,15 @@ class SceneryWindow final : public Window
             const auto lastTabIndex = GetMaxTabCountInARow() == MaxTabsPerRow ? MaxTabsPerRow - 1 : _tabEntries.size() - 1;
             const auto lastTabWidget = &widgets[WIDX_SCENERY_TAB_1 + lastTabIndex];
             windowWidth = std::max<int32_t>(windowWidth, lastTabWidget->right + 3);
+
+            if (_tabEntries.back().IsAll())
+            {
+                auto allTabWidget = &widgets[WIDX_SCENERY_TAB_1 + _tabEntries.size() - 1];
+                allTabWidget->left = windowWidth - TabWidth - 6;
+                allTabWidget->right = windowWidth - 7;
+                allTabWidget->top = InitTabPosY;
+                allTabWidget->bottom = InitTabPosY + TabHeight;
```

Comment:
The y position never changes, so this can be omitted.

Reply:
I wonder if an empty suggestion works. Let's see:
```suggestion
```

Reply:
The y position can change following #19373. Since the new tab is always at the end of the list, it may need it's vertical positioning adjusted.

Reply:
Omitting these two lines will give you this
![image](https://user-images.githubusercontent.com/3720978/220759807-9d556dc6-a55d-4d95-85fd-09a92cb0c7d0.png)

Reply:
Right, because this widget index is not guaranteed to keep the same index. That makes sense.